src/autoqasm/types/types.py

```diff
@@ -96,7 +96,9 @@ def __init__(
 
 
 class ArrayVar(oqpy.ArrayVar):
-    def __init__(self, *args, annotations: str | Iterable[str] | None = None, **kwargs):
+    def __init__(
+        self, init_expression, *args, annotations: str | Iterable[str] | None = None, **kwargs
```

Comment:
minor: can you add an appropriate type hint for `init_expression`? maybe `Iterable`?

Reply:
Added the type hint. I do think `Iterable` would suffice since it seems to encapsulate all cases (including multi-dimensional arrays).